spec/lib/miq_expression_spec.rb

```diff
@@ -3415,4 +3415,59 @@
       expect(result.map(&:first)).to include("VM and Instance : Id")
     end
   end
+
+  describe ".extract_key" do
+    subject { described_class.new(nil) }
+    let(:main_attribute) { Vm.arel_table[:id] }
+    let(:second_attribute) { GuestApplication.arel_table[:vm_or_template_id] }
+    let(:eq_node) { Arel::Nodes::Equality.new main_attribute, second_attribute }
+    let(:eq_other_node) { Arel::Nodes::Equality.new GuestApplication.arel_table[:type], Arel::Nodes::SqlLiteral.new("abc") }
+    let(:eq_bind_node) { Arel::Nodes::Equality.new GuestApplication.arel_table[:type], Arel::Nodes::BindParam.new("") }
+
+    it "handles equals clauses" do
+      query = subject.send(:extract_key, eq_node, main_attribute, second_attribute)
+      expect(query).to eq(nil)
+    end
+
+    it "handles equals bind clauses" do
+      query = subject.send(:extract_key, eq_node, main_attribute, second_attribute)
+      expect(query).to eq(nil)
+    end
+
+    it "handles reverse equals clauses" do
```

Comment:
```ruby
# Note: second and third args are the same for .extract_key
.where("'guest_applications'.vm_or_template_id = 'vms'.id")
```

Reply:
I was just putting stuff in them to see that it worked.

do you want me to actually build up different equal clauses?